### ISSUE TITLE
Error when specifying explicit underlying type on unscoped enums

### ISSUE THREAD

**Issue by pborsutzki** [shader-slang/slang#3959]
Hi,

the following code produces an error in Slang 2024.1.7, when compiling it via `slangc.exe -entry computeMain -stage compute -target spirv-asm -emit-spirv-directly`

```c++
[UnscopedEnum]
enum Foo : uint32_t {
    Bar = 42
};

[shader("compute")]
[numthreads(1, 1, 1)]
void computeMain()
{}
```

The reported error is:
```
shader.slang(6): fatal error 39999: cyclic reference 'Foo'.
enum Foo : uint32_t {
     ^~~
```

The error is gone, when removing the explicit type specification or the `[UnscopedEnum]` attribute.

Thanks for having a look!

**Comment by ArielG-NV** [comment]
to note what seems to be the issue for future debugging of issue:

Assume enum called `ourEnum`.

moduleScope --> _lookUpDirectAndTransparentMembers --> visitEnum [`ourEnum`] --> ensureDecl [uint32_t] ReadyForLookupResolve --> declToExclude is empty when resolving (likely the issue, should exclude `ourEnum` by this point?)  --> _lookUpInScopes [uint32_t is moduleScope, so it is in same scope as 'ourEnum'?] --> _lookUpDirectAndTransparentMembers --> visitEnum ['ourEnum'], causes failure

**Comment by csyonghe** [comment]
Fixed in #3962
